### PR TITLE
Added support for new file types in publishing request tickets

### PR DESCRIPTION
**Pull request by ketilaa**
... and fixed some flaky tests

### CODE REVIEW

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java [hunk 1]
```diff
@@ -158,64 +181,121 @@ private RuntimeException throwException(String message, Failure<?> failure) {
     }
 
     private void publishPublication(PublishingRequestCase latestUpdate, UserInstance userInstance) {
-        attempt(() -> resourceService.publishPublication(userInstance, latestUpdate.getResourceIdentifier()))
-            .orElseThrow(fail -> throwException(
-                String.format(PUBLISHING_ERROR_MESSAGE, latestUpdate.getResourceIdentifier()), fail));
+        attempt(
+                        () ->
+                                resourceService.publishPublication(
+                                        userInstance, latestUpdate.getResourceIdentifier()))
```
Comment (truhacevkir) [inline publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java:187]: Is it google code style, or stil unitStyle?

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java [hunk 2]
```diff
@@ -158,64 +181,121 @@ private RuntimeException throwException(String message, Failure<?> failure) {
...
+                        associatedArtifact ->
+                                publishFileIfApproved(associatedArtifact, publishingRequestCase))
+                .toList();
     }
 
-    private AssociatedArtifact publishFileIfApproved(AssociatedArtifact associatedArtifact,
-                                                     PublishingRequestCase publishingRequestCase) {
+    private AssociatedArtifact publishFileIfApproved(
+            AssociatedArtifact associatedArtifact, PublishingRequestCase publishingRequestCase) {
         if (associatedArtifact instanceof UnpublishedFile unpublishedFile) {
```
Comment (truhacevkir) [inline publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java:228]: Nice, 'if' statement can be replaced with 'switch' statement

**Review by truhacevkir — COMMENTED** [review]
Looks good, it seems like the only place we miss is PublishingRequestResolver which persists/updates publishing request and filesForApproval when making publication update.
One of methods which persists new publishingRequest on file update:

```
private void createPublishingRequestOnFileUpdate(Publication oldImage, Publication newImage) {
        if (containsNewPublishableFiles(oldImage, newImage)) {
            persistPendingPublishingRequest(oldImage, newImage);
        }
    }
```

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java (hunk 1 above)
Comment (ketilaa) [reply publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java:187]: I have the google code style enabled. Surprised by the amount of changes it make to the formatting.

Comment (truhacevkir) [reply publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java:187]: Should we add config for codestyle @LarsV123 implemented in search-api here as well? Cause for now the most of us still use NVA code style and not google code style.

File: publication-rest/src/main/java/no/unit/nva/publication/update/PublishingRequestResolver.java [hunk 3]
```diff
@@ -58,23 +63,24 @@ private boolean userCanManagePublishingRequests() {
...
     private static List<File> getUnpublishedFiles(Publication publication) {
-        return publication.getAssociatedArtifacts()
-                   .stream()
-                   .filter(UnpublishedFile.class::isInstance)
-                   .map(UnpublishedFile.class::cast)
-                   .collect(Collectors.toList());
+        return publication.getAssociatedArtifacts().stream()
+                .filter(File.class::isInstance)
+                .map(File.class::cast)
+                .filter(File::needsApproval)
```
Comment (truhacevkir) [inline publication-rest/src/main/java/no/unit/nva/publication/update/PublishingRequestResolver.java:82]: Nice

**Review by truhacevkir — APPROVED** [review]

**Comment by truhacevkir** [comment]
I think it should be tested via frontend/api's as well.

**Pull request merged**